### ISSUE TITLE
24.0.9 FVT: The Gradle dev command is taking longer than expected on the initial run.

### ISSUE THREAD

**Issue by aparnamichael** [OpenLiberty/liberty-tools-intellij#951]
When we run the start command for the first time in a Gradle project, it takes longer than expected and gives the impression that the dev mode is stuck. In the terminal, I can see the below message after dev mode starts.

`BUILD SUCCESSFUL in 4m 35s`

**Comment by TrevCraw** [comment]
Hi @aparnamichael , is this occurring on every OS?

**Comment by turkeylurkey** [comment]
No, but I have not cleared the Gradle cache. Is it spending time downloading the version of Gradle specified in the gradle wrapper?

**Comment by aparnamichael** [comment]
@turkeylurkey May be the Gradle download is taking a while since it's the first time, but from the terminal, it might seem like the dev mode is stuck. 

We both are using physical machines, not VMs for FVT testing.

**Comment by anusreelakshmi934** [comment]
I attempted to reproduce this issue across all three OS. On both Mac and Linux, the issue did not occur, and the process started without any noticeable delay. On Mac, it initiated within 32 seconds, while on Linux, it took approximately 1 minute and 15 seconds.
**Mac**
<img width="696" alt="image" src="https://github.com/user-attachments/assets/a8cdc1cd-59ed-49cf-a0f7-be7f5c815edb">
**Linux**
<img width="696" alt="image" src="https://github.com/user-attachments/assets/22ddb163-a4ba-4643-8a4e-b9d5a2928721">

On Windows, it took approximately 4 minutes for the server to fully start. It appears that the machine is downloading and installing the required features during this time. Since the Gradle project logs provide limited information during dev mode startup, it might give the impression that the dev mode is stuck, even though it is actively running and installing features. However, restarting the project doesn’t seem to cause this delay again, likely because the features are retrieved from the cache.
<img width="848" alt="image" src="https://github.com/user-attachments/assets/ea17c28b-ecd4-4929-a7a2-1d0313b76fbc">

**Comment by turkeylurkey** [comment]
Can you reproduce this issue from the command line without using IntelliJ? E.g. cd to the directory you see in the terminal output and then type the command you see in the terminal output. If not, try starting IntelliJ in one window and going back to your project in a different terminal and reproduce outside of IntelliJ. Maybe there is a lack of memory or some other resource.

**Comment by turkeylurkey** [comment]
Also could you state the kind of CPU, memory and disk that you are using?

**Comment by vaisakhkannan** [comment]
Now when I tried to reproduce this issue, it no longer occurs. I tried by opening in intellij itself. it only took around 1m 44s. Seems like it is intermittent.
![image](https://github.com/user-attachments/assets/af3242f8-7200-4ce9-810b-32738c677b81)

**Comment by vaisakhkannan** [comment]
> Also could you state the kind of CPU, memory and disk that you are using?

This is the device specification of windows.
![image](https://github.com/user-attachments/assets/110e5458-dd6f-4215-9ce2-f17f8fbe8cd1)

**Comment by anusreelakshmi934** [comment]
Now when I tried to reproduce this issue, it no longer occurs. I tried by opening in intellij itself. it only took around 1m 44s. Seems like it is intermittent.
![image](https://github.com/user-attachments/assets/74734259-45e8-4a0f-ac05-9ba2b2486c14)

**Comment by anusreelakshmi934** [comment]
> Also could you state the kind of CPU, memory and disk that you are using?

This is the device specification of windows.
![image](https://github.com/user-attachments/assets/61fb9341-d31f-4629-992b-f8edcb4e5d14)

**Comment by anusreelakshmi934** [comment]
As suggested by @turkeylurkey , I cleared the gradle cache and started dev mode in a fresh project. But was not able to reproduce the issue again. The dev mode started around 1m 40s.

**Comment by gilbysunil14** [comment]
Note: The progress bar in gradle is always shown as 0% because it is a never ending process until we stop the server.

**Comment by gilbysunil14** [comment]
Found a reasonable way to reproduce this issue:
1. Run the following commands in project terminal in the below specified order
- ~/.gradle/caches/
- ~/.gradle/wrapper/
- ./gradlew clean
2. Start liberty project in dev mode

By printing the debug logs found that the time spend by gradle is for downloading the feature dependencies

<img width="1660" alt="Image" src="https://github.com/user-attachments/assets/d9871b7b-4f3b-450d-9040-4577a592ede2" />

**Comment by turkeylurkey** [comment]
We should discuss closing this issue.

**Comment by TrevCraw** [comment]
@turkeylurkey I don't follow. Why would we close if we can reproduce?

**Comment by TrevCraw** [comment]
@gilbysunil14 , when you say "Start liberty project in dev mode", how do you do that?

Have you tried this outside of IntelliJ, just using the "gradle libertyDev" command in a terminal?

**Comment by turkeylurkey** [comment]
@TrevCraw Maybe I'm missing something but hasn't he identified that gradle is downloading dependencies and installing features? Things we just have to wait for?

**Comment by gilbysunil14** [comment]
> [@gilbysunil14](https://github.com/gilbysunil14) , when you say "Start liberty project in dev mode", how do you do that?
> 
> Have you tried this outside of IntelliJ, just using the "gradle libertyDev" command in a terminal?

Yes I have. The results were the same. Maybe for some the time takes more is because of the network connection. From what I can infer this time purely depends on the bandwidth of the network.

**Comment by gilbysunil14** [comment]
@TrevCraw, I have tried using the parameter `skipInstallFeature` while starting the dev mode to see if the initial time spend on installing the features can be skipped. But the parameter seems to work when dev mode is started on an already existing Liberty runtime installation. Which means the inital time spend on downloading dependencies and installing the features cannot be skipped using this parameter.

**Comment by anusreelakshmi934** [comment]
> Found a reasonable way to reproduce this issue:
> 
> 1. Run the following commands in project terminal in the below specified order
> 
> * rm -rf ~/.gradle/caches/
> * rm -rf ~/.gradle/wrapper/
> * ./gradlew clean
> 
> 2. Start liberty project in dev mode
> 
> By printing the debug logs found that the time spend by gradle is for downloading the feature dependencies
> 
> <img alt="Image" width="1660" src="https://private-user-images.githubusercontent.com/178545629/414629183-d9871b7b-4f3b-450d-9040-4577a592ede2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA2NjgyNTIsIm5iZiI6MTc0MDY2Nzk1MiwicGF0aCI6Ii8xNzg1NDU2MjkvNDE0NjI5MTgzLWQ5ODcxYjdiLTRmM2ItNDUwZC05MDQwLTQ1NzdhNTkyZWRlMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyN1QxNDUyMzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YWY4MDQ2ODhkZTU2NDI0YTRlMmUwYTFjNDMyOWIwZmFhNjQ4YWI5MjQxZDYyYWNjOWU0ODAzNDllMjk2OWM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ZuH3DwYIzrrCCQmT4sfuo-wXNWMnlnMaCoV5mxvfwyc">
> It took around 3m 6sec to start the server. Attaching log for reference.
> 
> [gradleLog.log](https://github.com/user-attachments/files/18864045/gradleLog.log)

By doing these mentioned steps, I was able to reproduce this issue in my environment also
